<!-- Phidgets IMU launch file -->

<launch>
	<!-- Nodelet manager -->
	<node
		pkg="nodelet"
		type="nodelet"
		name="imu_nodelet_manager"
		args="manager"
	/>

	<!-- IMU Driver -->
	<node
		pkg="nodelet"
		type="nodelet"
		name="PhidgetsImuNodelet"
		args="load phidgets_imu/PhidgetsImuNodelet imu_nodelet_manager">

		<param name="frame_id" value="imu"/>
		<param name="period" value="16"/>  <!-- rates: 4 8 16 24 32 ... 1000 [ms] -->
		<param name="angular_velocity_stdev" value="0.000349056"/>
		<param name="linear_acceleration_stdev" value="0.002943"/>
		<param name="magnetic_field_stdev" value="0.001658"/>

		<!-- Compass correction params -->
		<!--
		<param name="cc_mag_field" value="0.52859"/>
		<param name="cc_offset0" value="0.03921"/>
		<param name="cc_offset1" value="0.19441"/>
		<param name="cc_offset2" value="-0.03493"/>
		<param name="cc_gain0" value="1.81704"/>
		<param name="cc_gain1" value="1.81028"/>
		<param name="cc_gain2" value="2.04819"/>
		<param name="cc_t0" value="0.00142"/>
		<param name="cc_t1" value="-0.03591"/>
		<param name="cc_t2" value="0.00160 "/>
		<param name="cc_t3" value="-0.05038"/>
		<param name="cc_t4" value="-0.03942"/>
		<param name="cc_t5" value="-0.05673"/> -->
	</node>

	<!-- IMU Orientation Filter -->
	<node
		pkg="nodelet"
		type="nodelet"
		name="ImuFilterNodelet"
		args="load imu_filter_madgwick/ImuFilterNodelet imu_nodelet_manager" >

		<param name="world_frame" value="enu"/>
		<param name="use_mag" value="false"/>
		<param name="use_magnetic_field_msg" value="false"/>
		<param name="fixed_frame" value="odom"/>
		<param name="publish_tf" value="false"/>
		<param name="orientation_stddev" value="0.01"/>
	</node>

</launch>
